### ISSUE TITLE
API - search endpoint returns 500 response code when using all_ids option (possibly documentation bug)

### ISSUE THREAD

**Issue by walkerdb** [archivesspace/archivesspace#393]
According to the `/search` [endpoint documentation](https://archivesspace.github.io/archivesspace/doc/file.API.html#get-search), it requires one of these three options to be set in the request url: `page` (int), `id_set` (list), or `all_ids` (boolean).

When I perform a search with `page=1` and json search parameters, it works fine. When I try the same with `all_ids=true` instead of `page` I receive a 500 error response saying there's an undefined method for "nil.NilClass".

@djpillen took a look through some of the source to see where the issue might be and found the following:

The `search` endpoint controller [marks it as paginated](https://github.com/archivesspace/archivesspace/blob/master/backend/app/controllers/search.rb#L54).
The documentation-building code [gives all paginated endpoints the text that states it requires **either** `page`, `id_set`, or `all_ids`](https://github.com/archivesspace/archivesspace/blob/master/backend/scripts/endpoint_doc.rb#L16-L22).
However, the `search` endpoint model [only handles requests with `page` set](https://github.com/archivesspace/archivesspace/blob/master/backend/app/model/search.rb#L19), and has no functionality defined for any of the other two options. Most other endpoints set to paginated use the [handle_listing function](https://github.com/archivesspace/archivesspace/blob/4c26d82b1b0e343b7e1aea86a11913dcf6ff5b6f/backend/app/lib/crud_helpers.rb#L48) to handle the other options, but since search queries solr directly, it looks like that might have been overlooked.

**Comment by cfitz** [comment]
Right, I think this is a documentation bug. The search endpoints are a bit unique...will fix this.

**Comment by lmcglohon** [comment]
@walkerdb I believe this has been implemented. I cannot replicate the problem so I am closing this issue. Please let me know if you think it needs to be reopened.

**Comment by sdm7g** [comment]
Yes -- this is still a problem. 
Calling /search without any parameters indicates that all_ids is a valid parameter:

```
{
    "error": {
        "page": [
            "Parameter required but no value provided"
        ],
        "id_set": [
            "Parameter required but no value provided"
        ],
        "all_ids": [
            "Parameter required but no value provided"
        ]
    }
}

```

API docs say:
- Boolean   all_ids – Return a list of all object ids

But /search?all_ids=true  ( with or without a query parameter ) returns:
`{
    "error": "undefined method '-' for nil:NilClass"
}`

/search/repositories  shows the same issues. 

BTW: The new feature where issues shows you related issues worked nicely. 
I searched open issues and didn't see anything, but when I started to enter this as a new issue, 
related issues popped up with this closed one.
